pkg/controller/controller.go

```diff
@@ -170,20 +175,19 @@ func (h *Handler) syncLocalVolumeProvider(instance *localv1.LocalVolume) error {
 		})
 	}
 	o.Status.Generations = children
-	o.Status.ObservedGeneration = &o.Generation
 	o.Status.State = operatorv1.Managed
-	condition := operatorv1.OperatorCondition{
-		Type:               operatorv1.OperatorStatusTypeAvailable,
-		Status:             operatorv1.ConditionTrue,
-		Message:            "Ready",
-		LastTransitionTime: metav1.Now(),
-	}
-	newConditions := []operatorv1.OperatorCondition{condition}
-	o.Status.Conditions = newConditions
-	err = h.apiClient.syncStatus(instance, o)
-	if err != nil {
-		klog.Errorf("error syncing status : %v", err)
-		return fmt.Errorf("error syncing status %v", err)
+	o = h.addSuccessCondition(o)
+
+	if !equality.Semantic.DeepEqual(instance.Status, o.Status) {
+		// A generation update can cause frivolous updates to LocalVolume object
+		// we are only going to update generation when something in LocalVolume object
+		// changes.
+		o.Status.ObservedGeneration = &o.Generation
```

Comment:
I am not sure about this change. When someone changes `.spec` (= `generation` is incremented), `.status` should be changed too so users can see that the operator is doing something about their change. If `observedGeneration` is stuck at an older value it shows that something is wrong with the operator.

Reply:
So, `.Generation`  is a meta field which is marked as `readonly` and only api-server increments it each time an object is saved.

So what is happening is - lets say there was no other update to `LocalVolume` object and we set `Status.ObservedGeneration` to current value of `.Generation` - but act of updating `.Status.ObservedGeneration`  will cause `.Generation` to be incremented. So this results in a infinite loop where `LocalVolume` is always being updated.  Among other things - this causes unnecessary churn  in api-server and I am not sure it is necessary.

Reply:
So in a nutshell - at any point you observe `LocalVolume` - its `ObservedGeneration` will be one number behind `.Generation` .. no matter what we do.

Reply:
> act of updating .Status.ObservedGeneration will cause .Generation to be incremented

That should not happen. `status` should be a subresource and its change then does not bump `metadata.generation`.

> Among other things - this causes unnecessary churn in api-server and I am not sure it is necessary.

It is one of ways how to detect unresponsive operator.

Reply:
I managed to update `status` this way:
https://github.com/openshift/csi-operator/blob/master/pkg/controller/csidriverdeployment/csidriverdeployment_controller.go#L501

And CRD gest status subresource here:
https://github.com/openshift/csi-operator/blob/master/deploy/prerequisites/01_crd.yaml#L14-L15

Reply:
Yeah - I did not mark `status` as a subresource in the CRD.  I could do that.

Reply:
Since I am not using version of operator-sdk you were using.. this will be slightly more complicated I think..